qiita_db/base.py

```diff
@@ -128,28 +130,36 @@ def _check_id(self, id_):
         the other classes. However, still defining here as there is only one
         subclass that doesn't follow this convention and it can override this.
         """
-        self._check_subclass()
-
-        conn_handler = SQLConnectionHandler()
-
-        return conn_handler.execute_fetchone(
-            "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
-            "{0}_id=%s)".format(self._table), (id_, ))[0]
-
-    def __init__(self, id_):
+        sql = "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE {0}_id=%s)".format(
+            self._table)
+        trans.add(sql, [id_])
+        # The value that we want is the result of the last SQL query,
+        # and it is stored in the first value of the first row
+        return trans.execute()[-1][0][0]
```

Comment:
No. As stated, it would do full indexing of `[-1][0][0]` or `[-1][0]` if that makes it more widely applicable. Looking through the pull requests, this would cover a large fraction of the use cases. This does not affect other cases where `trans.execute()[-1]` is still applicable, but cleans up the longer, more convoluted indexing. Plus, two reviewers now have found the indexing to be an issue, so it is best to centralize it into the execute function with a detailed explanation.